plugins/catalog-backend-module-gitea/src/providers/GiteaEntityProvider.ts

```diff
@@ -0,0 +1,266 @@
+    }
+  }
+
+  private createScheduleFn(
+    taskRunner: SchedulerServiceTaskRunner,
+  ): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class: GiteaEntityProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(
+              `${this.getProviderName()} refresh failed, ${error}`,
+              error as Error,
+            );
+          }
+        },
+      });
+    };
+  }
+
+  async refresh(logger: LoggerService): Promise<void> {
+    if (!this.connection) {
+      throw new Error('Gitea discovery connection not initialized');
+    }
+    let allRepos: any[] = [];
+
+    try {
+      // Fetch all repos for an organization
+      const OrgRepoApiUrl = `${getGiteaApiUrl(this.integration.config)}orgs/${
+        this.config.organization
+      }/repos`;
+      let page = 1;
+      let hasMoreData = true;
+
+      while (hasMoreData) {
+        const url = `${OrgRepoApiUrl}?page=${page}`;
+        const response = await fetch(url, {
+          method: 'GET',
+          ...getGiteaRequestOptions(this.integration.config),
+        });
+
+        if (!response.ok) {
+          throw new Error(
+            `Failed to fetch page ${page} of repos, status: ${response.status}`,
+          );
+        }
+
+        const projectsPage = (await this.parseGiteaJsonResponse(
+          response,
+        )) as GiteaProjectQueryResult;
+
+        if (!Array.isArray(projectsPage) || projectsPage.length === 0) {
+          hasMoreData = false;
+        } else {
+          allRepos = allRepos.concat(projectsPage);
+          page++;
+        }
+      }
+    } catch (e) {
+      throw new Error(
+        `Failed to list Gitea projects for organization ${this.config.organization}, ${e}`,
+      );
+    }
+
+    // Filter for repos that have catalog-info.yaml at root
+    const { default: pLimit } = await import('p-limit');
+    const limit = pLimit(5);
+    const validRepos: string[] = [];
+
+    await Promise.all(
+      allRepos.map(repo =>
+        limit(async () => {
+          if (repo.empty) {
+            logger.warn(`Repo ${repo.html_url} is empty, skipping`);
+            return;
+          }
+          const contentsApiUrl = `${getGiteaApiUrl(
+            this.integration.config,
+          )}repos/${this.config.organization}/${repo.name}/contents/${
+            this.config.catalogPath
+          }`;
+
+          const res = await fetch(contentsApiUrl, {
+            method: 'GET',
+            ...getGiteaRequestOptions(this.integration.config),
+          });
+          if (res.ok) {
+            validRepos.push(repo.html_url);
+          } else {
+            logger.warn(
```

Comment:
This cannot reasonably be a warning right? 🤔 That's got to be very very common when scraping and lead to log noise. _At most_ debug (or nothing at all) I'd say